optuna/study.py

```diff
@@ -125,13 +129,15 @@ def objective(trial):
                 Note that if you set the flag to :obj:`False`, you shouldn't mutate
                 any fields of the returned trial. Otherwise the internal state of
                 the study may corrupt and unexpected behavior may happen.
+            state:
+                Trial states to filter on. If :obj:`None`, include all states.
```

Comment:
Following the discussion about `state`'s type. I'd like you to update this to mention that `state` (may be `states`) is required to be a container.

Reply:
Isn't it clear from the type?

Reply:
Absolutely.